saleor/checkout/complete_checkout.py

```diff
@@ -859,6 +872,11 @@ def complete_checkout_post_payment_part(
 ) -> Tuple[Optional[Order], bool, dict]:
     action_required = False
     action_data: Dict[str, str] = {}
+    code = None
+    voucher = order_data.get("voucher")
+
+    if voucher and checkout_info.checkout.voucher_code:
+        code = voucher.codes.filter(code=checkout_info.checkout.voucher_code).first()
```

Comment:
This is the next place where we are asking for code from `voucher.codes` relation. Maybe we should change the `checkout_info` to keep the `voucher_code` instance instead of `voucher` instance

Reply:
In my opinion, Saleor should store both in `checkout_info`. `voucher` is required during the recalculation process, and `code` is required for the validation process.